src/app/zap-templates/helper-chip.js

```diff
@@ -65,10 +52,102 @@ function chip_header() {
   */`;
 }
 
+function isString(type) {
+  return type == 'CHAR_STRING' || type == 'OCTET_STRING' || type == 'LONG_CHAR_STRING' || type == 'LONG_OCTET_STRING';
+}
+
+function isNullable(type, isArray) { return isString(type) || isArray; }
+
+function asValueIfNotPresent(type) {
+  var promise = templateUtil.ensureZclPackageId(this)
+                    .then((packageId) => {
+                      switch (type) {
+                      case 'CHAR_STRING':
+                      case 'OCTET_STRING':
```

Comment:
This function could use a small rework. Moving out the string type from the switch case would prevent executing promises when not needed.

```
if (isNullable(type, isArray)) {
    return 'NULL'
}
var promise = ....etc